### PR TITLE
Easter egg

### PR DESCRIPTION
**Pull request by ortiza5**
Tried to implement that snow overlay as an easter egg. I made it so if you started a theme name with "[letitsnow](https://pbs.twimg.com/media/EI-0_w1XUAAgRa8.jpg)", the overlay should be added.

Unfortunately, I couldn't get the video to work. It is posted here as a draft in case you have any ideas about how to make it work. I already tried changing the `src` to the [imgur link](https://i.imgur.com/6LKzfxG.mp4), but that resulted in a paused video overlay.

Since, it is just a bit of fun, there is no need to spend too much time on it.

### CODE REVIEW

**Comment by luetage** [comment]
For me the autoplay gets blocked, although I have set it to »allow« in privacy settings. I also get an additional error about the service worker failing to register. The video gets added, but it doesn’t play.

**Comment by luetage** [comment]
I converted it to a gif, I think this could be a chance to add it more seamlessly. But the resulting file is over 6MB in size, 6 times the size of the whole mod. Don’t know whether that’s a good idea. Generally I think it would be a good easter egg, we just have to find another solution.

**Comment by ortiza5** [comment]
Tried playing around with a .WEBP, but I couldn't get a smooth framerate because all the little snow flakes really increased the bitrate. None of the encoding options I tried in Screen2Gif could smooth it out.

Switched it over to HTML/CSS. It now generates 80 randomized snowflakes and adds them to the DOM to be animated with CSS.

It would also be nice if each flake wasn't a circle (maybe there is a way to change them with CSS?), but this overall is definitely a cleaner approach than the background video. Anything you would change?

**Comment by ortiza5** [comment]
Changed it to only require the theme name to contain `letitsnow` when stripped of spaces and converted to lowercase.

**Comment by ortiza5** [comment]
Then we also have to consider how we are going to tease the easter egg on the forum. Figure most people won't be watching these pull request comments.

Options:
* Put "let it snow" in the release comments without context
* Post a GIF of it snowing an let people look at this repo to figure out the change
* Share a theme with "let it snow" in the name
* Post [this](https://pbs.twimg.com/media/EI-0_w1XUAAgRa8.jpg) picture with a cryptic message (maybe not the best option since one of the few people that want this easter egg is stardepp and the image is in English)
   * Could also just use an image of the guy in the image if we wanted to be extra cryptic, but that requires some knowledge of the meme or the US show SNL. I haven't seen more than a few clips of SNL, but I don't know how common knowledge of that skit is outside the US
* Or just say something interesting happens if you put "let it snow" in a theme name

**Comment by luetage** [comment]
Don’t think I would personally every use it, I don’t like moving things when I want to read something. But yeah, we can leave it like that. What I don’t understand is the second if statement when checking for the magic word. It only triggers when the name starts with letitsnow exactly, so why is it needed?

**Comment by luetage** [comment]
I think it’s a little confusing, starting the name with “letitsnow” is easy enough.

File: theme.js [hunk 1]
```diff
@@ -42,6 +78,10 @@ function loadTheme() {
         for (const [key, value] of Object.entries(colors)) {
           document.body.style.setProperty("--" + key, value);
         }
+        const strippedTheme = theme.split('_').join('').toLowerCase();
```
Comment (ortiza5) [inline theme.js:81]: > What I don’t understand is the second if statement when checking for the magic word. It only triggers when the name starts with letitsnow exactly, so why is it needed?

Are you looking at the most recent version? I had an if statement checking for the words at the beginning of the name, but I removed that to make it easier to trigger the easter egg.

This new check just makes it so all of these theme names should trigger it:
* "Let It Snow Theme"
* "Word let it snow"
* "Theme letitsnow 2"
* "leTitsnOw example"

File: theme.js [hunk 2]
```diff
@@ -42,6 +78,10 @@ function loadTheme() {
...
+        if (strippedTheme.includes("letitsnow")) {
+          loadEasterEgg();
+        }
```
Comment (ortiza5) [inline theme.js:84]: > What I don’t understand is the second if statement when checking for the magic word. It only triggers when the name starts with letitsnow exactly, so why is it needed?

Are you looking at the most recent version? I had an if statement checking for the words at the beginning of the name, but I removed that to make it easier to trigger the easter egg.

This new check just makes it so all of these theme names should trigger it:
* "Let It Snow Theme"
* "Word let it snow"
* "Theme letitsnow 2"
* "leTitsnOw example"

**Comment by luetage** [comment]
Do you know what might be better, setting up an event listener for the main input field (search field) and triggering the mod from there. It isn’t really attached to the theme anyway, because a scheduled or regular theme change won’t stop the snow until a reload. Easier to trigger and users don’t have to change theme names.

File: theme.js (hunk 2 above)
Comment (luetage) [reply theme.js:84]: Yeah, I probably looked at the wrong code.

**Comment by ortiza5** [comment]
Ok, that is a good idea, but maybe just put the listener on the page itself? Using the search field is similar to the Biscuit Mode activation, but I like the idea of being free to activate/deactivate it while focus is anywhere on the page.

I remember that fun old Easter Egg on on `vogue.co.uk` worked like that with the Konami Code to spawn fashionable velociraptors. https://www.youtube.com/watch?v=VBmcv_iQzWg

**Comment by ortiza5** [comment]
Ah, looks like you have made changes already. Are you able to push commits to the branch of this pull request?

Otherwise, I can go ahead and close this PR since your branch already has the commits.

**Comment by luetage** [comment]
I made a pull request on your original branch, the changes aren’t in this pull request.

**Comment by ortiza5** [comment]
Weird, didn't get a notification that you made a pull request, just noticed your branch in the main repo. Will have to investigate my notification settings.

**Comment by luetage** [comment]
Are you on the west coast? We have pretty dissimilar sleep schedules.

And putting it on the page itself would be an option of course. But we would be listening to every single keystroke a user makes, which is a bit of a waste of computing time and also invasive, but more importantly you can’t see what you type or input, which in case of number of snowflakes could be bothersome for people who can’t type fluently. But don’t know.

**Comment by ortiza5** [comment]
Yeah, I am in the Mountain Standard Time zone (UTC−7). I also seem to be most productive in the evenings, so it is easy for me to really throw my sleep schedule off when I have free time. What time zone are you in?

Those are good arguments for not putting the listener on the page. Google might even have some objections to it.

**Comment by luetage** [comment]
Central European Time UTC+01:00

**Comment by luetage** [comment]
About the announcement, I’ll leave that up to you how to “spoil” the news. I probably won’t include it in the announcement. I’ll probably be making some changes to the dark mode integration still, maybe I can get the dark skin to be compatible with custom themes so it doesn’t matter which skin is in place when loading a custom theme. I made a feature request for a theme toggle in the header, but I’m not certain it will be implemented and it won’t be implemented in time, because I intend to push this to webstore before 2023.

**Comment by ortiza5** [comment]
Oops, checked with larger numbers of snow flakes and saw a horizontal gap in the flakes. The issue is with this line: https://github.com/luetage/vivaldi_forum_mod/blob/5d38e7425dbc3814d21288bd37f50814f8510ae8/default.js#L167

It should actually be this to span the whole range of the animations:
`animation-delay: ${genRand(-12, 0, 1)}s, ${genRand(-4, 0, 1)}s;`

Not sure it is worth a whole new pull request. Could you just make the change?

**Comment by luetage** [comment]
done

**Pull request merged**